feat: solana code-fence features in main cp-12.18.0

aganglada opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

* Moves code-fence solana features to `main` build

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/32284?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

aganglada committed:
feat: solana code-fence features in main

javiergarciavera committed:
feat: moved Solana tests to regular build for release

javiergarciavera committed:
feat: added FF for createSolanaaCcount

javiergarciavera committed:
Merge branch 'main' of github.com:MetaMask/metamask-extension into feat/solana-in-main

javiergarciavera committed:
feat: fixed state for solana

javiergarciavera committed:
fix: fix carousel tests

javiergarciavera committed:
feat: fix conflicts

cmd-ob committed:
fix: add account type filtering filtering for accout sync tests

cmd-ob committed:
Merge branch 'feat/solana-in-main' of github.com:MetaMask/metamask-extension into feat/solana-in-main

javiergarciavera committed:
feat: update import path

javiergarciavera committed:
feat: fix lint

javiergarciavera committed:
feat: missing file

darioAnongba committed:
Merge branch 'main' into feat/solana-in-main

adonesky1 committed:
Merge branch 'main' into feat/solana-in-main

darioAnongba committed:
Merge branch 'main' into feat/solana-in-main

mathieuartu reviewed (dismissed):
Identity changes LGTM, summoning @cmd-ob for a second pair of eyes!

javiergarciavera committed:
Merge branch 'main' into feat/solana-in-main

javiergarciavera committed:
fix: fixing some flakiness on testHelpers file

javiergarciavera committed:
Merge branch 'main' into feat/solana-in-main

adonesky1 committed:
fix lint + flaky solana wallet standard test

adonesky1 committed:
another delay

darioAnongba committed:
Merge branch 'main' into feat/solana-in-main

aganglada committed:
Merge branch 'main' into feat/solana-in-main

javiergarciavera committed:
feat: fix multichain api e2e test flakiness

javiergarciavera committed:
fix: carousel number of cards updated

javiergarciavera committed:
Merge branch 'main' of github.com:MetaMask/metamask-extension into feat/solana-in-main

javiergarciavera committed:
fix: switch number of cards in the carousel back to 7

zone-live committed:
Merge branch 'main' into feat/solana-in-main

javiergarciavera committed:
fix: fix carousel flakiness

javiergarciavera committed:
Merge branch 'feat/solana-in-main' of github.com:MetaMask/metamask-extension into feat/solana-in-main

javiergarciavera committed:
fix: fixed race condition on carousel test

javiergarciavera committed:
feat: posible fix

cmd-ob committed:
fix: ensure account type is consistently checked in backup and sync tests

hjetpoluru committed:
Merge branch 'main' into feat/solana-in-main

javiergarciavera committed:
feat: fixed carousel once more time

adonesky1 committed:
lint

adonesky1 committed:
delay

javiergarciavera committed:
feat: fixing flakiness

javiergarciavera committed:
feat: increased timeout

javiergarciavera committed:
feat: reduced timeout to speed up tests

javiergarciavera committed:
feat: increased timeout

javiergarciavera committed:
feat: added no custom network client found

javiergarciavera committed:
feat: fixed more flakiness

seaona committed:
test: skip solana (#32545)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/32545?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

javiergarciavera committed:
Merge branch 'main' into feat/solana-in-main

javiergarciavera committed:
feat: added message

javiergarciavera committed:
feat: lint

javiergarciavera committed:
fix: skip multichain test case
